homeassistant/helpers/device_registry.py

```diff
@@ -46,29 +47,33 @@ def async_get_device(self, identifiers: str, connections: tuple):
         """Check if device is registered."""
         for device in self.devices:
             if any(iden in device.identifiers for iden in identifiers) or \
-                    any(conn in device.connection for conn in connections):
+                    any(conn in device.connections for conn in connections):
                 return device
         return None
 
     @callback
-    def async_get_or_create(self, *, connection, identifiers, manufacturer,
-                            model, name=None, sw_version=None):
+    def async_get_or_create(self, *, config_entry, connections, identifiers,
+                            manufacturer, model, name=None, sw_version=None):
         """Get device. Create if it doesn't exist."""
-        device = self.async_get_device(identifiers, connection)
+        device = self.async_get_device(identifiers, connections)
 
         if device is not None:
+            if config_entry not in device.config_entries:
+                device.config_entries.add(config_entry)
+                self.async_schedule_save()
             return device
 
         device = DeviceEntry(
```

Comment:
We should not create a new device if it has no connections or identifiers, as it can never be retrieved again

Reply:
Should identifiers be mandatory or should it check and fail for any of them?

Reply:
I guess you're right that at least one is mandatory, I am fine with ignoring it when it doesn't match and just print a warning